### PR TITLE
SHARE-434 Add studio overview

### PR DESCRIPTION
**Pull request by schaubes**
- list all public studios
- list private studios of user

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
No additional description

### Tests - additional information
No additional information

### CODE REVIEW

File: assets/styles/custom/studio.scss [hunk 1]
```diff
@@ -1,6 +1,88 @@
 @use "sass:math";
 @import "../base/variables";
 
+$list-item-padding-overflow-x: 1.5rem;
+
+/* -- STUDIOS OVERVIEW -- */
```
Comment (dmetzner) [inline assets/styles/custom/studio.scss]: Am besten direkt ein eigenes File, wenn die Trennung so klar ist

Overview, Detail, ..

File: assets/styles/custom/studio.scss [hunk 2]
```diff
@@ -1,6 +1,108 @@
...
+@use "@material/list/mdc-list";
+@use "@material/menu-surface/mdc-menu-surface";
+@use "@material/menu/mdc-menu";
 @import "../base/variables";
 
+$list-item-padding-overflow-x: 1.5rem;
+
+/**
+ * STUDIOS OVERVIEW
+ */
```
Comment (dmetzner) [inline assets/styles/custom/studio.scss]: In so einem Fall gern in eigenes File

File: src/Application/Controller/Studio/StudioController.php [hunk 3]
```diff
@@ -59,17 +73,94 @@ public function studioDetails(Request $request): Response
...
+
+    $name = trim((string) $request->request->get('name', ''));
+    $description = trim((string) $request->request->get('description', ''));
+    if ('' === $name) {
+      return new JsonResponse(['message' => 'arguments invalid'], Response::HTTP_BAD_REQUEST);
+    }
+
+    $existingStudio = $this->studio_manager->findStudioByName($name);
+    if ($existingStudio) {
+      return new JsonResponse(['message' => 'studio name is already taken'], Response::HTTP_BAD_REQUEST);
```
Comment (dmetzner) [inline src/Application/Controller/Studio/StudioController.php]: Http 409 wär hier auch eine möglichkeit

File: templates/Studio/studio_new.html.twig [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{% extends 'Default/base.html.twig' %}
+
+{% block head %}
+    {{ encore_entry_link_tags('studios_overview') }}
+{% endblock %}
+
+{% block body %}
+
+    <h1>Create New Studio</h1>
```
Comment (dmetzner) [inline templates/Studio/studio_new.html.twig]: Texte mit dem Translation system?

**Review by dmetzner — APPROVED** [review]

File: templates/Studio/studio_new.html.twig (hunk 4 above)
Comment (schaubes) [reply templates/Studio/studio_new.html.twig]: Wie füg ich hier neue values dazu? einfach ins 'en' file schreiben?

Comment (dmetzner) [reply templates/Studio/studio_new.html.twig]: ja genau :)

**Pull request merged**